docs/tooling/package-metadata.md

```diff
@@ -0,0 +1,192 @@
+
+```js
+"scripts": {
+  "start": "backstage-cli package start",
+  "build": "backstage-cli package build",
+  "lint": "backstage-cli package lint",
+  "test": "backstage-cli package test",
+  "clean": "backstage-cli package clean",
+  "prepack": "backstage-cli package prepack",
+  "postpack": "backstage-cli package postpack"
+}
+```
+
+### `configSchema`
+
+The Backstage configuration schema for the package, as described in the [defining configuration](../conf/defining.md) section.
+
+### `backstage`
+
+This field is a collection of Backstage specific metadata fields. It is required for all Backstage packages, and any package that defines this field is considered to be part of the Backstage ecosystem. All sub-fields of this collection are defined below.
+
+### `backstage.role`
+
+This field defines the role of the package in the Backstage ecosystem. It can affect both the build process and runtime behavior, and signals the intended usage of the package to consumers. You can read more about this field in the [package roles](./cli/02-build-system.md#package-roles) section.
+
+### `backstage.pluginId`
+
+For any package that is part of a plugin, this field should be set to the plugin ID. This is the same ID as you would pass to the `createPlugin`, `createBackendPlugin`, or `createBackendModule` functions in the implementation of the package. It is also the same ID as the one described in the [name](#name) section.
+
+This field can be generated by the `backstage-cli repo fix --publish` command. The plugin ID will be inferred from the package [name](#name) and [role](#backstagerole). If the package name is not actually part of a plugin but still has `plugin-*` in its name, you can set this field to be explicitly `null`.
+
+The presence of this field is checked by the `backstage-cli package prepack` command, which is used to prepare a package for publishing. You can read more about this requirement in the section on [metadata for published packages](#metadata-for-published-packages).
+
+### `backstage.pluginPackages`
+
+For any package that is part of a plugin, this field should be set to a list of all packages that are directly part of the same plugin. This includes frontend and backend plugin packages as well as related libraries, but not modules.
+
+This field can be generated by the `backstage-cli repo fix --publish` command. It will list all packages with the same [plugin ID](#backstagepluginid) in the workspace.
+
+The presence of this field is checked by the `backstage-cli package prepack` command, which is used to prepare a package for publishing. You can read more about this requirement in the section on [metadata for published packages](#metadata-for-published-packages).
+
+```js title="Example usage of the backstage.pluginPackages field"
+{
+  "name": "@backstage/plugin-catalog",
+  "backstage": {
+    "role": "frontend-plugin",
+    "pluginId": "catalog",
+    "pluginPackages": [
```

Comment:
Still tempted to have this one renamed honestly - just TOO similar to `pluginPackage` :) `relatedPackages` or `bundlePackages` would be good perhaps. Not as part of this PR of course, but

Reply:
Let's discuss it tomorrow 😁